d3ploy/solver.py

```diff
@@ -166,7 +169,17 @@ def get_asc_key_list(dicti):
     """
     key_list = [' '] * len(dicti.values())
     sorted_caps = sorted(dicti.values(), reverse=True)
+    # store previous indx list to prevent
+    # error caused by prototypes with same capacity
+    prev_indx_list = []
```

Comment:
Anyway, this whole section could use some meaningful names. This whole function (beginning https://github.com/ergs/d3ploy/pull/137/files#diff-e29e462b4265bec544e3cd52c7114600R155) seems to be reinventing the OrderedDict. I think this whole function should be deleted and should be replaced by an ordered dict upstream. Up to you, but that would be my recommendation.